src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -114,28 +119,102 @@ static void checkArg(string arg, string name)
                 Arguments = appArgs,
                 WorkingDirectory = workingDir,
                 GuiApp = guiApp,
-                EnvironmentVariables = envVariables
+                EnvironmentVariables = envVariables,
+                IsRunWithExit = metricType == MetricType.DotnetWatch ? false : true
             };
-            
+
+            ProcessHelper secondTestProcess = metricType == MetricType.InnerLoop || metricType == MetricType.InnerLoopMsBuild ? TestProcess : null;
+
+            //Create wait funcs for steady state and post-compilation
+            Func<Process, bool> waitForSteadyState = metricType == MetricType.DotnetWatch ? (Proc) =>
+            {
+                StringBuilder output = new StringBuilder();
+                DataReceivedEventHandler stdOutProcessor = (s, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        output.AppendLine(e.Data);
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                Proc.OutputDataReceived += stdOutProcessor;
+                Proc.BeginOutputReadLine();
```

Comment:
Should this also read stderr?

Reply:
I am not sure on this. Maybe we could use output on stderr to indicate that we have reached a failure point, but I am not sure that that would always hold. I am having a hard time imagining any scenario where we would want to inspect and know the contents of stderr as part of a correctly functioning test, but maybe there is one? Maybe it is just good to have from a logging perspective.

Reply:
I just have it read it and write it out to the console for now.